api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -479,18 +486,21 @@ type NodePoolCondition struct {
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
 }
 
-// Taint is as v1 Core but without TimeAdded.
+// taint is as v1 Core but without TimeAdded.
 // https://github.com/kubernetes/kubernetes/blob/ed8cad1e80d096257921908a52ac69cf1f41a098/staging/src/k8s.io/api/core/v1/types.go#L3037-L3053
 type Taint struct {
-	// Required. The taint key to be applied to a node.
+	// +required
+	// key is the taint key to be applied to a node.
 	Key string `json:"key"`
-	// The taint value corresponding to the taint key.
+	// value is the taint value corresponding to the taint key.
 	// +optional
 	// +kubebuilder:validation:Pattern:=`^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$`
```

Comment:
Prefer to use CEL so that you can add a custom message to explain the validation/regex